src/screens/Server.js

```diff
@@ -87,26 +122,64 @@ class Server extends React.Component {
       return;
     }
 
-    this.setState({ loading: true, errorMessage: '', selectedServer: newServer });
-    const currentServer = hathorLib.helpers.getServerURL();
-    // Update new server in local storage
-    hathorLib.wallet.changeServer(newServer);
-    hathorLib.versionApi.getVersion((data) => {
-      // If the user selected a testnet server, we show a modal to confirm the operation
-      if (data.network !== 'mainnet') {
+    this.setState({
+      loading: true,
+      errorMessage: '',
+      selectedServer: newBaseServer,
+      selectedWsServer: newWsServer,
+    });
+
+    const currentServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseUrl() : 
+      hathorLib.config.getServerUrl();
+
+    const currentWsServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseWsUrl() :
+      '';
+
+    // Update new server in storage and in the config singleton
+    this.props.wallet.changeServer(newBaseServer);
+
+    // We only have a different websocket server on the wallet-service facade, so update the config singleton
+    if (this.props.useWalletService) {
+      this.props.wallet.changeWsServer(newWsServer);
+    }
+
+    try {
+      const versionData = await this.props.wallet.getVersionData();
+      if (versionData.network !== 'mainnet') {
+        const network = versionData.network;
+        let selectedNetwork = '';
+
+        // Network might be 'testnet-golf' or 'testnet-charlie'
+        if (network.startsWith('testnet')) {
+          selectedNetwork = 'testnet';
+        } else {
+          selectedNetwork = network;
```

Comment:
Even though we prepare for values others than `^testnet*` and `mainnet`, I don't see clearly how they would be accepted on the rest of the process here.

Is this allowed now, or a future enhancement?

Reply:
Solved on the `utils/version` file :+1: